client/blocks/comment-detail/index.jsx

```diff
@@ -188,32 +184,27 @@ export class CommentDetail extends Component {
 	}
 }
 
-const mapStateToProps = ( state, { commentId, siteId } ) => {
-	//const comment = getComment( state, siteId, commentId );
-	const comment = mockComment;
-
-	return {
-		authorAvatarUrl: get( comment, 'author.avatar_URL' ),
-		authorDisplayName: get( comment, 'author.name' ),
-		authorEmail: get( comment, 'author.email' ),
-		authorId: get( comment, 'author.ID' ),
-		authorIp: get( comment, 'author.ip' ), // TODO: not available in the current data structure
-		authorIsBlocked: get( comment, 'author.isBlocked' ), // TODO: not available in the current data structure
-		authorUrl: get( comment, 'author.URL' ),
-		authorUsername: get( comment, 'author.nice_name' ),
-		commentContent: comment.content,
-		commentDate: comment.date,
-		commentId: commentId,
-		commentIsApproved: 'approved' === comment.status,
-		commentIsLiked: comment.i_like,
-		commentIsSpam: 'spam' === comment.status,
-		commentIsTrash: 'trash' === comment.status,
-		postAuthorDisplayName: get( comment, 'post.author.name' ),
-		postTitle: get( comment, 'post.title' ),
-		postUrl: get( comment, 'post.link' ),
-		repliedToComment: comment.replied, // TODO: not available in the current data structure
-		siteId: siteId,
-	};
-};
+const mapStateToProps = ( state, ownProps ) => ( {
+	authorAvatarUrl: get( ownProps, 'author.avatar_URL' ),
+	authorDisplayName: get( ownProps, 'author.name' ),
+	authorEmail: get( ownProps, 'author.email' ),
+	authorId: get( ownProps, 'author.ID' ),
+	authorIp: get( ownProps, 'author.ip' ), // TODO: not available in the current data structure
+	authorIsBlocked: get( ownProps, 'author.isBlocked' ), // TODO: not available in the current data structure
+	authorUrl: get( ownProps, 'author.URL' ),
+	authorUsername: get( ownProps, 'author.nice_name' ),
+	commentContent: ownProps.content,
+	commentDate: ownProps.date,
+	commentId: ownProps.commentId,
+	commentIsApproved: 'approved' === ownProps.status,
+	commentIsLiked: ownProps.i_like,
+	commentIsSpam: 'spam' === ownProps.status,
+	commentIsTrash: 'trash' === ownProps.status,
+	postAuthorDisplayName: get( ownProps, 'post.author.name' ),
+	postTitle: get( ownProps, 'post.title' ),
+	postUrl: get( ownProps, 'post.link' ),
+	repliedToComment: ownProps.replied, // TODO: not available in the current data structure
+	siteId: ownProps.siteId,
+} );
```

Comment:
> For example, a comment cannot be both spam and trash, so why does our data object allow that invalid representation?

That makes sense to remove the redundant booleans and stick with a status enum. Though we'll likely want to keep`isLiked`. We can have an approved comment, that is not liked for example.

> I was about to remove them altogether, but I didn't want to increase the size of this PR even more.

I wouldn't mind if you refactored a bit here. We don't need to be as cautious when the components are not in use anywhere in prod.